### PR TITLE
Feat: Add enable param in Database and Collection

### PR DESCRIPTION
**Pull request by 2002Bishwajeet**
## What does this PR do?

Adds `enabled` parameter while creating/updating Database and Collection

## Related PRs and Issues

nil

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Comment by 2002Bishwajeet** [comment]
cc @abnegate

**Review by abnegate — COMMENTED** [review]
Looks good :+1: just some small formatting comments. I've re-run the CI to check tests again too (the ones that failed are flaky and unrelated to your changes)

**Review by abnegate — COMMENTED** [review]
Please add some more tests to cover all other updates endpoints too :pray:

**Review by abnegate — COMMENTED** [review]
Nice work, just a couple of formatting issues

**Comment by abnegate** [comment]
Just realized we also need to add the checks to ensure that only the console can access these routes if a database is disabled, you can see an example of the same check for storage buckets [here](https://github.com/appwrite/appwrite/blob/b86ce2d2339fab4803ab9a10b939d4330551c6b0/app/controllers/api/storage.php#L367). 

We will need this check for all of the create/read/update/delete routes for collections and documents.

Send me a DM if you need any clarification :smile:

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -193,16 +193,14 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
         ->setType(DATABASE_TYPE_CREATE_ATTRIBUTE)
         ->setDatabase($db)
         ->setCollection($collection)
-        ->setDocument($attribute)
-    ;
+        ->setDocument($attribute);
```
Comment (abnegate) [inline app/controllers/api/databases.php:196]: Please remove all the formatting updates, it makes the PR harder to review as it adds a lot of unrelated changes to the diff

**Review by abnegate — COMMENTED** [review]
Nice, please also add tests to `DatabasesConsoleClientTest` that ensures that the console can still access these routes for a disabled database

File: app/controllers/api/databases.php [hunk 2]
```diff
@@ -781,9 +787,9 @@ function updateAttribute(
     ->inject('dbForProject')
     ->action(function (string $databaseId, array $queries, string $search, Response $response, Database $dbForProject) {
 
-        $database = Authorization::skip(fn () => $dbForProject->getDocument('databases', $databaseId));
+        $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
 
-        if ($database->isEmpty()) {
+        if ($database->isEmpty() || (!$database->getAttribute('enabled'))) {
```
Comment (abnegate) [inline app/controllers/api/databases.php:792]: Let's add the admin mode check here too

File: app/controllers/api/databases.php [hunk 3]
```diff
@@ -836,9 +842,9 @@ function updateAttribute(
     ->inject('dbForProject')
     ->action(function (string $databaseId, string $collectionId, Response $response, Database $dbForProject) {
 
-        $database = Authorization::skip(fn () => $dbForProject->getDocument('databases', $databaseId));
+        $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
 
-        if ($database->isEmpty()) {
+        if ($database->isEmpty() || !$database->getAttribute('enabled')) {
```
Comment (abnegate) [inline app/controllers/api/databases.php:847]: Let's add the admin mode check here too

File: app/controllers/api/databases.php [hunk 4]
```diff
@@ -975,9 +981,9 @@ function updateAttribute(
     ->inject('events')
     ->action(function (string $databaseId, string $collectionId, string $name, ?array $permissions, bool $documentSecurity, bool $enabled, Response $response, Database $dbForProject, Event $events) {
 
-        $database = Authorization::skip(fn () => $dbForProject->getDocument('databases', $databaseId));
+        $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
 
-        if ($database->isEmpty()) {
+        if ($database->isEmpty() || !$database->getAttribute('enabled')) {
```
Comment (abnegate) [inline app/controllers/api/databases.php:986]: Let's add the admin mode check here too

File: app/controllers/api/databases.php [hunk 5]
```diff
@@ -1040,9 +1046,9 @@ function updateAttribute(
     ->inject('deletes')
     ->action(function (string $databaseId, string $collectionId, Response $response, Database $dbForProject, Event $events, Delete $deletes) {
 
-        $database = Authorization::skip(fn () => $dbForProject->getDocument('databases', $databaseId));
+        $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
 
-        if ($database->isEmpty()) {
+        if ($database->isEmpty() || !$database->getAttribute('enabled')) {
```
Comment (abnegate) [inline app/controllers/api/databases.php:1051]: Let's add the admin mode check here too

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 6]
```diff
@@ -50,6 +51,38 @@ public function testCreateCollection(): array
...
+            'name' => 'TvShows',
+            'permissions' => [
+                Permission::read(Role::any()),
+                Permission::create(Role::any()),
+                Permission::update(Role::any()),
+                Permission::delete(Role::any()),
+            ],
+            'documentSecurity' => true,
+        ]);
+
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:82]: Let's add assertions for list/get/update/delete collection in console mode when the database is disabled as well

Comment (2002Bishwajeet) [reply tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:80]: Done

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 7]
```diff
@@ -50,7 +50,131 @@ public function testCreateCollection(): array
...
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $collection['headers']['status-code']);
+        $this->assertEquals('Movies', $collection['body']['name']);
+        $this->assertEquals($moviesCollectionId, $collection['body']['$id']);
+        $this->assertTrue($collection['body']['enabled']);
+    }
+
+
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:131]: Let's separate functions by only one blank line

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 8]
```diff
@@ -50,7 +50,131 @@ public function testCreateCollection(): array
...
+            'enabled' => false
+        ]);
+
+        $this->assertEquals(200, $collection['headers']['status-code']);
+        $this->assertEquals('Movies Updated', $collection['body']['name']);
+        $this->assertEquals($moviesCollectionId, $collection['body']['$id']);
+        $this->assertFalse($collection['body']['enabled']);
+    }
+
+
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:159]: Let's separate functions by only one blank line

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 9]
```diff
@@ -50,7 +50,131 @@ public function testCreateCollection(): array
...
+     * @param array $data
+     */
+    public function testGetCollection(array $data)
+    {
+        $databaseId = $data['databaseId'];
+        $moviesCollectionId = $data['moviesId'];
+        /**
+         * Test When database is disabled but can still call get collection
+         */
+
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:119]: Let's have a blank line before the comment instead of after

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 10]
```diff
@@ -50,7 +50,131 @@ public function testCreateCollection(): array
...
+    }
+
+
+    /**
+     * @depends testCreateCollection
+     * @param array $data
+     */
+    public function testUpdateCollection(array $data)
+    {
+
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:138]: Let's remove this blank line

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 11]
```diff
@@ -50,7 +50,131 @@ public function testCreateCollection(): array
...
+        $databaseId = $data['databaseId'];
+        $moviesCollectionId = $data['moviesId'];
+        /**
+         * Test When database is disabled but can still call update collection
+         */
+
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:144]: Let's have a blank line before the comment instead of after

File: tests/e2e/Services/Databases/DatabasesConsoleClientTest.php [hunk 12]
```diff
@@ -50,7 +50,131 @@ public function testCreateCollection(): array
...
+    /**
+     * @depends testCreateCollection
+     * @param array $data
+     */
+    public function testDeleteCollection(array $data)
+    {
+        $databaseId = $data['databaseId'];
+        $tvShowsId = $data['tvShowsId'];
+        /**
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesConsoleClientTest.php:168]: Let's add a blank line before the comment

**Review by abnegate — COMMENTED** [review]
Some small formatting issues and we're good! Make sure to run `composer format` or `docker run -itv $PWD:/app composer format` to auto-fix any issues possible 👍

File: tests/e2e/Services/Databases/DatabasesCustomServerTest.php [hunk 13]
```diff
@@ -506,6 +544,54 @@ public function testListCollections()
         ]);
 
         $this->assertEquals(409, $response['headers']['status-code']);
+        return
+            [
+                'databaseId' => $databaseId,
+                'collectionId' => $test1['body']['$id'],
+            ];
```
Comment (abnegate) [inline tests/e2e/Services/Databases/DatabasesCustomServerTest.php:551]:
```suggestion
        return [
            'databaseId' => $databaseId,
            'collectionId' => $test1['body']['$id'],
        ];
```

**Review by abnegate — APPROVED** [review]
